cpp/benchmarks/join/join_benchmark.cu

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <benchmark/benchmark.h>
+#include <vector>
+#include <cudf/cudf.h>
+#include <utilities/error_utils.hpp>
+#include <tests/utilities/column_wrapper.cuh>
+
+#include "generate_input_tables.cuh"
+
+
+template<typename key_type, typename payload_type>
+static void join_benchmark(benchmark::State& state)
+{
+    const gdf_size_type build_table_size {(gdf_size_type) state.range(0)};
+    const gdf_size_type probe_table_size {(gdf_size_type) state.range(1)};
+    const gdf_size_type rand_max_val {build_table_size * 2};
+    const double selectivity = 0.3;
+    const bool is_build_table_key_unique = true;
+
+    // Generate build and probe tables
+
+    cudf::test::column_wrapper<key_type> build_key_column(build_table_size);
+    cudf::test::column_wrapper<key_type> probe_key_column(probe_table_size);
+
+    generate_input_tables<key_type, gdf_size_type>(
+        (key_type *)build_key_column.get()->data, build_table_size,
+        (key_type *)probe_key_column.get()->data, probe_table_size,
+        selectivity, rand_max_val, is_build_table_key_unique
+    );
+
+    cudf::test::column_wrapper<payload_type> build_payload_column(
+        build_table_size,
+        [] (gdf_index_type row_index) {
+            return row_index;
+        }
+    );
+
+    cudf::test::column_wrapper<payload_type> probe_payload_column(
+        probe_table_size,
+        [] (gdf_index_type row_index) {
+            return row_index;
+        }
+    );
+
+    CUDA_TRY(cudaGetLastError());
+    CUDA_TRY(cudaDeviceSynchronize());
+
+    std::vector<gdf_column*> build_table {build_key_column.get(), build_payload_column.get()};
+    std::vector<gdf_column*> probe_table {probe_key_column.get(), probe_payload_column.get()};
+
+    // Setup join parameters and result table
+
+    gdf_context ctxt = {
+        0,                     // input data is not sorted
+        gdf_method::GDF_HASH   // hash based join
+    };
+
+    int columns_to_join[] = {0};
+
+    const int nresult_cols = build_table.size() + probe_table.size() - 1;
+
+    std::vector<gdf_column> result_table(nresult_cols);
+    std::vector<gdf_column *> col_ptrs(nresult_cols);
+
+    for (size_t icol = 0; icol < result_table.size(); icol++) {
+        col_ptrs[icol] = &result_table[icol];
+    }
+
+    // Benchmark the inner join operation
+
+    CUDA_TRY(cudaDeviceSynchronize());
```

Comment:
Again, no need to synchronize here, it will just make benchmarking take longer...

Reply:
Resolved in https://github.com/rapidsai/cudf/pull/2105/commits/10d4e20c9ab010c429d853125fb4cffe821554fd.